pkg/provider/azure_loadbalancer.go

```diff
@@ -3063,48 +3063,84 @@ func (az *Cloud) reconcileSecurityGroup(
 	}
 
 	var (
-		disableFloatingIP                                = consts.IsK8sServiceDisableLoadBalancerFloatingIP(service)
-		lbIPAddresses, _                                 = iputil.ParseAddresses(lbIPs)
-		lbIPv4Addresses, lbIPv6Addresses                 = iputil.GroupAddressesByFamily(lbIPAddresses)
-		additionalIPv4Addresses, additionalIPv6Addresses = iputil.GroupAddressesByFamily(additionalIPs)
-		backendIPv4Addresses, backendIPv6Addresses       []netip.Addr
+		dstIPv4Addresses, dstIPv6Addresses         []netip.Addr
+		dstIpv4AddressPrefix, dstIpv6AddressPrefix []netip.Prefix
 	)
-	{
-		// Get backend node IPs
-		lb, lbFound, err := az.getAzureLoadBalancer(ctx, lbName, azcache.CacheReadTypeDefault)
-		{
+
+	if az.IsLBBackendPoolTypePodIP() {
+		if !az.RetrievedClusterPodCidr {
```

Comment:
Generally, this would block the security boundary of the Kubernetes cluster. The cluster identity should not have the permission to manage itself.

Reply:
let's use new cloud config options and ask the provisioning service (e.g. AKS or capz) to setup the CIDRs on changing.